autorest/azure/auth/auth.go

```diff
@@ -741,9 +741,15 @@ type MSIConfig struct {
 
 // ServicePrincipalToken creates a ServicePrincipalToken from MSI.
 func (mc MSIConfig) ServicePrincipalToken() (*adal.ServicePrincipalToken, error) {
-	spToken, err := adal.NewServicePrincipalTokenFromManagedIdentity(mc.Resource, &adal.ManagedIdentityOptions{
-		ClientID: mc.ClientID,
-	})
+	mio := adal.ManagedIdentityOptions{}
+
+	if strings.Contains(mc.ClientID, "/Microsoft.ManagedIdentity/userAssignedIdentities/") {
```

Comment:
@chlowell do we need to support this in `azidentity`?

Reply:
`azidentity` supports specifying a user-assigned identity by its resource ID but doesn't support passing a resource ID as a client ID. I don't want to support that ☺️

Reply:
True.  I was more wondering about support client and resource IDs, then I remembered that we [do](https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/azidentity/managed_identity_credential.go#L21-L22) in a much cleaner way.

Reply:
So we do support client resource IDs at the [adal](https://github.com/Azure/go-autorest/blob/master/autorest/adal/token.go#L766-L774) level.  Apparently we never exposed this in the `auth` package.  Do you think this is a safe workaround for that?  The other option would be to add an `IdentityResourceID` field to the `MSIConfig` type.

Reply:
I would expose the field. This approach is probably safe, but I think we should maintain the distinction between client and resource IDs. Users generally know the difference and I expect they would find it unintuitive to assign a resource ID to a `ClientID` field--that looks like a bug.